google-cloud-spanner/src/main/java/com/google/cloud/spanner/Options.java

```diff
@@ -123,6 +123,15 @@ public static ListOption pageToken(String pageToken) {
     return new PageTokenOption(pageToken);
   }
 
+  /**
+   * Specifying this will cause the transaction to be committed when executing the DML statement. A
+   * requirement is that the transaction should have a single action which is a DML statement. This
+   * option is automatically specified if Connection.isAutocommit() is true.
+   */
+  public static InlineCommitOption inlineCommitOption() {
```

Comment:
Would it in some way be possible to make this package private, or otherwise not accessible for normal users of the client library? It seems like an option that might be difficult for users to use in the correct way when using a normal transaction runner. If we want to support this in the client library, I think it would be better to add a separate method for it. Something like `DatabaseClient#executeUpdateAutocommit(..)`. That way the user does not have any way of accidentally executing multiple statements in a transaction with autocommit enabled.

Reply:
That was our initial plan. But I couldn't find a way to implement it.
What we need is for Connection to pass to DatabaseClient the info that Connection.isAutocommit() is true. But they are in different packages. Any function we introduce to DatabaseClient that enables Connection to call, the users can call too. Thiago and I discussed this too. Anything I missed?

Reply:
Ah, yes, that is true.

What are your thoughts on adding one of the following methods to `DatabaseClient` instead of allowing this option for `TransactionRunner`:
- `DatabaseClient#executeUpdate`: Allows the user to execute one single update statement without having to create a `TransactionRunner`. The method would use the inline commit option internally. The connection API could also call this method when it wants to execute a single update statement in autocommit mode.
- `DatabaseClient#singleUseTransaction()`: Returns a single-use read/write transaction that can be used to execute one update. Works otherwise in the same way as the above `executeUpdate` method. The idea behind this would be to keep it in line with how users execute single reads (`singleUse().executeQuery(..)`). It would also be a good entry point if we want to support inline commit for Batch DML: `singleUseTransaction().batchUpdate(statement1, statement2, ...)`

The disadvantage of the above is obviously that it does not allow any select-then-update semantics when using inline commit, but I'm not sure that is a big miss, considering that this option is mainly intended for supporting autocommit in the Connection API.

@thiagotnunes Do you have any thoughts on the above?

Reply:
We raised this point before, but could not find any good enough solution. However, from reading your suggestions I quite like the introduction of a `DatabaseClient#singleUseTransaction` as it fits very nicely with the other methods that we have. This way we could remove the `Options#inlineCommit` and prevent users from having multiple statements within an auto-commit enabled transaction.

@dzhang-bap could you change the implementation to introduce this interface method instead? When you do, provide a default implementation that just throws an error like [here](https://github.com/googleapis/java-spanner/blob/7d6816f1fd14bcd2c7f91d814855b5d921ba970d/google-cloud-spanner/src/main/java/com/google/cloud/spanner/TransactionContext.java#L107-L109), in order to prevent breaking changes.

Reply:
I love the idea!!! I've introduced a new class SingleDmlTransaction, and added DatabaseClient#singleDmlTransaction(). Note that there exists a class called SingleUseTransaction so I want to use a different name. Besides, "SingleDmlTransaction" seem more accurate because we only allow a DML here, not query.